evals/api.py

```diff
@@ -23,85 +24,139 @@
 logger = logging.getLogger(__name__)
 
 
-def completion_query(
-    model_spec: ModelSpec,
-    prompt: Union[OpenAICreatePrompt, OpenAICreateChatPrompt, Prompt],
-    **kwargs,
-) -> tuple[dict, Union[OpenAICreatePrompt, OpenAICreateChatPrompt], dict]:
-    """
-    Query the API for a completion.
+class ReturnType(ABC):
+    def __init__(self, raw_data: Any, prompt: Any, metadata: Dict[str, Any] = None):
+        self.raw_data = raw_data
+        self.prompt: str = prompt
+        self.completions: List[str] = self.extract_completions()
```

Comment:
`self.extract_completions()` is called twice, once in the constructor here and another in the eval classes (e.g. Match line 45.

I think we can make `completion` or `completions` a property of `ReturnType` and the eval class just access them.